### ISSUE TITLE
PieceCID is not a CID, it confuses people, and should be renamed PieceCommitment

### ISSUE THREAD

**Issue by hannahhoward** [filecoin-project/specs#855]
In conversations with non technical people, PieceCID is repeatedly confused with PayloadCID, and it often takes long explanations to help people understand it is not the root of the IPLD Graph of the data you want to store, but instead the value we call "CommP" -- which stands for PieceCommitment.

Moreover, the value is not infact a CID -- it lacks multibase information, it does not conform to any CID spec, and moreover is not itself ultimately a hash of bytes!

I propose, and I understand this might make for a sizable find/replace, that PieceCID be renamed PieceCommitment, to clearly identify that it is not in fact a CID, and definitely not the root of the IPLD graph of data a client is intending to store.

**Comment by porcuquine** [comment]
I believe that @jbenet has previously made it clear that we do need use an actual CID. The proofs library does not handle CIDs directly, and doesn't plan to — but this is because we assume the caller will manage any packaging and unpackaging which may be needed.

**Comment by hannahhoward** [comment]
absolutely @porcuquine -- I just think it shouldn't have CID in the name since that is a specification that exists and the value does not conform to that specification.

**Comment by mishmosh** [comment]
Related: #172

**Comment by anorth** [comment]
I believe @jbenet's intention is that it *should be a real CID* on chain, should carry multibase information etc, be the root of the DAG that is the piece merkle tree. But, at the interface to the proofs library (in implementation), the raw hash extracted and passed.

**Comment by porcuquine** [comment]
Yes, @anorth describes my understanding. Also, somewhat confusingly, the *spec* does use CIDs in interfaces. The reason I mentioned the actual behavior of the `rust-fil-proofs` was to clarify this point (and that we knowingly don't follow the reference description at that interface boundary — but without invalidating the need for the CID to manifest somewhere).

**Comment by anorth** [comment]
Quoth @jbenet out of band:
> the CID prefix (multibase, multicodec, version) needs to be added to the raw part (commP)
